docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+
+Our global scheduler does the scheduling/placement of a VM/Container to the granularity of a cluster. To avoid a huge DB, we should not store every node's allocable capacity information in the global DB. But if we only store each cluster's accumulated allocable capacity, "_quantity check_" cannot be done properly. This is because even a cluster's accumulated allocable capacity is larger than a VM/Container's quantity requirement, the cluster may have no single node which has enough resource to host/run the VM/Container. An "average per node allocable capacity" is a better way. But "_quantity check_" against the "average per node allocable capacity" is not good enough either. As shown in Figure 2, the new VM requires 128M memory and Cluster1's "average per node available memory resource" is 110M memory. Although Node 3 in Cluster 1 has 256M, which meets the VM's memory requirement, the "_quantity check_" against Cluster 1 will fail and Cluster 1 will be incorrectly filtered out of the candidate list.
+
+Therefore, instead of doing "_quantity check_" against a cluster's "average per-node allocable capacity", we should do "_quantity check_" against a cluster's "largest available node's resource". But checking against only the largest available node of a cluster is not enough. This is because a VM/Container's resource requirement spans more than one resource type. A node having the largest available vCPU may have little available memory. So, we need to collect the resource information of several largest available nodes from each cluster (a few with the largest CPU, a few with the largest memory, etc.). Let's call these nodes "potentially feasible nodes". Resource information of each cluster's "potentially feasible nodes" should be saved in the global resource database. "_quantity check_" should be done against a cluster's "potential feasible nodes" to see whether there is one node that has enough vCPU, memory, EIPs, disk volume to host/run the VM/Container. If so, the cluster is added to the candidate list.
+![image.png](/images/2.5.2.png)
+
+After the scheduling algorithm finds the candidate list of clusters for a VM/Container, it goes to the second phase---scoring phase.  In the scoring phase, the scheduling algorithm runs a set of functions to score each cluster in the candidate list and picks a cluster with the highest score.
+
+Factors that need to be considered for scoring include the following (more can be added in the future):
+
+1. average per node allocable capacity. The more allocable capacity, the higher the score. But a VM/Container's resource requirement spans more than one resource type. A cluster may have more "per node available memory resource" but less "per node available vCPU resource" than another cluster. How to define "more" or "less"? Section 2.5.2.1 describes the way to define this.
```

Comment:
Global scheduler do not  decision the placement of a  VM onto a host, I think we don’t need to consider the dimension of per node capacity.

Reply:
This is the scroing phase.  Although the global scheduler does not place a VM onto a node, a cluster that has high average per-node capacity is better than a cluster that has low average per-node capacity if the policy is to have even resource distribution across clusters. Of course, the average per-node capacity is just one factor in the final score and its weight can be adjusted.